doc/manual/src/contributing/experimental-flags.md

```diff
@@ -0,0 +1,89 @@
+This section aims at describing the notion of a Nix "experimental
+flag", and how it fits into the big picture of the developpement of Nix.
+
+# What are these xp flags?
+
+Since Nix 2.4, some features of Nix can be flagged as
+experimental.Conceptually, this means that these features are not
+considered as part of Nix proper, but are only included in the code to
+make their developpement and testing easier.Practically, these require
+an explicit opt-in, by adding the name of the experimental feature to
+the `experimental-features` configuration option. This is really handy
+as it makes it harder to accidentally use an experimental feature
+without knowing it.
+
+# When should an experimental flag be used
+
+A change in the Nix codebase should be guarded by an experimental flag
+if it is likely to either introduce a regression of any kind, or if it
+changes the external interface of Nix (be it the language, the cli or
+anything else that's considered as "public").
+
+For example, these changes need to be guarded by an experimental flag:
+
+-   Any language change (new builtin, syntactic or semantic change,
+    etc..),
+
+-   Any change to the (non-experimental) CLI interface
+
+-   A non-trivial internal refactoring that's likely to introduce some
+    bugs\[\^1\]
+
+\[\^1\]: Having to be weighted against the costs of course. If the
+refactoring changes a low-level interface and keeping both versions
+around essentially means duplicating half of the codebase, then it might
+not be worth doing that.
+
+# Life flow of an experimental flag
+
+Experimental features being, well, experimental, they all have to be
+treated on a case-by-case basis.However, the standard workflow for an
+experimental flag is as follows:
+
+-   The feature is merged, under an experimental flag (possibly over a
+    long period of time)
+
+-   It becomes part of a release
+
+-   After at least a full 6w release cycle, the feature becomes enabled
+    by default
+
+-   After at lesat another full 6w release cycle, the old behavior is
+    removed
+
+This is obviously just indicative. In particular the third step is only
+useful if the feature is susceptible to introduce some significant and
+hard to work-around regression (for example the addition of a new
+language builtin). Conversely, the second is useless if people have no
```

Comment:
This is unclear, because why would the addition of a new builtin cause a regression?

Reply:
Indeed. I think I first wrote the sentence the other way, so that was actually an exemple of something that’s unlikely to warrant this third step :upside_down_face: